src/blr/cli/filterclusters.py

```diff
@@ -54,9 +55,12 @@ def strip_barcode(pysam_read, tags_to_be_removed, removed_tags):
 
     # Remove tags
     for bam_tag in tags_to_be_removed:
-        removed_tags[bam_tag].add(pysam_read.get_tag(bam_tag))
-        # Strip read from tag
-        pysam_read.set_tag(bam_tag, None, value_type="Z")
+
+        if pysam_read.has_tag(bam_tag):
```

Comment:
Is this check needed? This function is only run `if no_mols and no_mols > threshold`, which shuold also mean it requires the read to have the tag right?

Reply:
Yes it is needed. Now when duplicate reads are also to be tagged with the MN for `no_mols` they will still miss the MI tag for molecule index. Thus they would raise an error here if not checked first.

Reply:
But if we add the MN tag, wouldn't it also make sense to also add the MI tag as well? 

Otherwise we'll be a bit inconsistent since duplicate reads will have some, but not all tags normal reads have. Basically I'm promoting either removing all or keeping all tags from the duplicate reads.

Reply:
You mean adding it but with a `None` value or such? Or maybe `-1` as it has to be integers? I guess that is the alternative. 

We do have read without the BX tag (or any other of our tags) so we are not entirely consistent in that case too.

Reply:
I mean if a duplicate read has a `BX` tag it should also get the `MN` and `MI` tag as all other reads (which will be identical to the non-duplicate read at that position presumably). 

So instead of having 
```
read1 ... <not duplicate> BX:Z:ATGC MN:I:<num_mol> MI:<ID>
read2 ... <duplicate> BX:Z:ATGC
```
we write 
```
read1 ... <not duplicate> BX:Z:ATGC MN:I:<num_mol> MI:<ID>
read2 ... <duplicate> BX:Z:ATGC MN:I:<num_mol> MI:<ID>
```
and rely downstream analysis on not looking at duplicates if they aren't relevant. 

Essentially, for all processing I believe duplicate reads should be treated the same (such as **adding** molecule information or **merging** barcode duplicate), while excluding them from all analysis step (such as **building** the molecules or **finding** the barcode duplicates).

Does that distinguishment make sense?